utils/utils.go

```diff
@@ -14,101 +14,238 @@ limitations under the License.
+	}
+
+	return nil
+}
+
+// Unzip a CSAR file
+func (c *CSARFile) Unzip(filePath string) error {
+	reader, err := zip.OpenReader(filePath)
 	if err != nil {
-		return nil, pkgerrors.Wrap(err, "Read CSAR file error")
+		return pkgerrors.Wrap(err, "CSAR file extracting error")
 	}
 
-	return rawBody, nil
+	for _, file := range reader.File {
+		path := filepath.Join(file.Name)
+
+		if file.FileInfo().IsDir() {
+			os.MkdirAll(path, file.Mode())
+			continue
+		}
+
+		fileReader, err := file.Open()
+		if err != nil {
+			return pkgerrors.Wrap(err, "CSAR file extracting error")
+		}
+		defer fileReader.Close()
+
+		targetFile, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, file.Mode())
+		if err != nil {
+			return pkgerrors.Wrap(err, "CSAR file extracting error")
+		}
+		defer targetFile.Close()
+
+		_, err = io.Copy(targetFile, fileReader)
+		if err != nil {
+			return pkgerrors.Wrap(err, "CSAR file extracting error")
+		}
+	}
+	return nil
 }
 
-// DownloadCSAR to download the CSAR files form URL and parse it
+// Delete a CSAR file
+func (c *CSARFile) Delete(path string) error {
+	err := os.RemoveAll(path)
+	if err != nil {
+		return pkgerrors.Wrap(err, "CSAR file delete error")
+	}
+	return nil
+}
+
+// CreateKubeObjectsFromCSAR to download the CSAR files from URL and extract it
 // to get deployment and service yamls
-var DownloadCSAR = func(url string) (*CSARData, error) {
-	rawFile, err := Download(url)
+var CreateKubeObjectsFromCSAR = func(csarID string, csarURL string) (*KubernetesData, error) {
+
+	// 1. Download CSAR file
+	err := CSAR.Download("csar_file.zip", csarURL)
 	if err != nil {
 		return nil, err
 	}
 
-	// This will be changed to whatever is actually present in a CSAR
-	csar := &CSARData{
-		CSARdata: rawFile,
+	// 2. Unzip CSAR file which will contain a deployment folder containining
+	// deployment.yaml and service.yaml.
+	err = CSAR.Unzip("csar_file.zip")
+	if err != nil {
+		return nil, err
 	}
 
-	err = csar.ParseDeploymentInfo()
+	kubeData := &KubernetesData{}
+
+	// 3. Read the deployment.yaml and service.yaml and set the deployment and
+	// service structs.
+	err = kubeData.ReadDeploymentYAML(csarID + "/deployment.yaml")
+	if err != nil {
+		return nil, err
+	}
+	err = kubeData.ReadServiceYAML(csarID + "/service.yaml")
 	if err != nil {
 		return nil, err
 	}
 
-	err = csar.ParseServiceInfo()
+	// 4. Delete extracted directory.
+	err = CSAR.Delete(csarID)
 	if err != nil {
 		return nil, err
 	}
 
-	return csar, nil
+	// 5. Delete CSAR file.
+	err = CSAR.Delete("csar_file.zip")
+	if err != nil {
+		return nil, err
+	}
+
+	// 6. Return kubeData struct for kubernetes client to spin things up.
+	return kubeData, nil
 }
 
-// CSARParser is an interface to parse both Deployment and Services
+// CSARKubeParser is an interface to parse both Deployment and Services
 // yaml files
-type CSARParser interface {
+type CSARKubeParser interface {
+	ReadDeploymentYAML(string) error
```

Comment:
Maybe it's possible to have only two methods (ReadYAML and ParseInfo) and discover the type of the object thru several attempts	

                err = c.ParseDeploymentInfo()
		if err != nil {
                        err = c.ParseServiceInfo()
                        if err != nil {
			    return err
                        }
		}